File: examify-backend/app/Models/Classroom.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Classroom extends Model
{
    use HasFactory;

    protected $fillable = [
        'teacher_id',
        'name',
        'description',
        'join_code',
        'is_meeting_active',
    ];

    public function teacher()
    {
        return $this->belongsTo(User::class, 'teacher_id');
    }

    public function students()
    {
        return $this->belongsToMany(User::class, 'classroom_students', 'classroom_id', 'student_id');
    }

    public function assessments()
    {
        return $this->hasMany(Assessment::class);
    }

    public function announcements()
    {
        return $this->hasMany(Announcement::class);
    }
}

File: examify_flutter/lib/shared/models/classroom.dart
import 'user.dart';

class Classroom {
  final int id;
  final String name;
  final String joinCode;
  final bool isMeetingActive;
  final String? description;
  final User? teacher;
  final int? studentsCount;

  Classroom({
    required this.id,
    required this.name,
    required this.joinCode,
    this.isMeetingActive = false,
    this.description,
    this.teacher,
    this.studentsCount,
  });

  factory Classroom.fromJson(Map<String, dynamic> json) {
    return Classroom(
      id: json['id'],
      name: json['name'],
      joinCode: json['join_code'] ?? '',
      isMeetingActive:
          json['is_meeting_active'] == 1 || json['is_meeting_active'] == true,
      description: json['description'],
      teacher: json['teacher'] != null ? User.fromJson(json['teacher']) : null,
      studentsCount: json['students_count'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'join_code': joinCode,
      'is_meeting_active': isMeetingActive,
      'description': description,
      'teacher': teacher?.toJson(),
      'students_count': studentsCount,
    };
  }
}
